src/ramanchada2/protocols/calibration/xcalibration.py

```diff
@@ -80,15 +80,12 @@ def process(
                     new_spe.x = self.model(new_spe.x)
                 if not np.all(np.diff(new_spe.x) > 0):
                     if self.nonmonotonic == "nan":
-                        new_spe.x = np.where(
-                            np.diff(new_spe.x, prepend=new_spe.x[0]) < 0,
-                            np.nan,
-                            new_spe.x,
-                        )
+                        new_spe.x = np.asarray(new_spe.x, dtype=float)
+                        is_nonmonotonic = np.diff(new_spe.x, prepend=new_spe.x[0]) < 0
```

Comment:
- we need *strictly* monotonically increasing function so Δx ≤ 0
- this approach does not work in general
```python
import numpy as np
x = np.array([30., 40, 32, 33, 34, 1, 2, 4, 5])
d = np.diff(x, prepend=[0])
x[d<0] = np.nan
x  # array([30., 40., nan, 33., 34., nan,  2.,  4.,  5.])
```
- if the result is not strictly monotonically increasing function, the result is wrong so the `self.nonmonotonic == 'error'` should be default. (it might already be. didn't follow)

Reply:
ok, will replace with <=0 
nans are  mostly to handle extrapolation, and nans in extrapolated areas are ok in most cases 
interpolation should be handled by pchip itseld

Reply:
this should be resolved now
after filling in with nans there is a final check and if still nonmonotonic, exception is raised